### ISSUE TITLE
magicsock: DERP and roaming address oscillation

### ISSUE THREAD

**Issue by crawshaw** [tailscale/tailscale#118]
morty's logs see a lot of:

```
magicsock: rx [pub:…e4Vj] from roaming address 207.108.197.248:41641, set as new priority
magicsock: rx [pub:…e4Vj] from known 127.3.3.40:1 (0), replaces roaming address 207.108.197.248:41641
magicsock: rx [pub:…e4Vj] from roaming address 207.108.197.248:41641, set as new priority
magicsock: rx [pub:…e4Vj] from known 127.3.3.40:1 (0), replaces roaming address 207.108.197.248:41641
magicsock: rx [pub:…e4Vj] from roaming address 207.108.197.248:41641, set as new priority
magicsock: rx [pub:…e4Vj] from known 127.3.3.40:1 (0), replaces roaming address 207.108.197.248:41641
```

According to CONTROL this is one of @danderson 's machines, which has somehow roamed onto a new port (CONTROL thinks it is on 207.108.197.248:1394). That's pretty interesting, I guess this is an example of us needing to STUN more.

But in magicsock is means we have a lot of log noise talking to it. Spray packets come in from both DERP and the roaming address, meaning in `AddrSet` we swap address priority a lot.

Should we make less log noise? Avoid replacing roaming addresses?

cc @danderson @bradfitz

**Comment by crawshaw** [comment]
Thinking about this in terms of launching DERP: packets still get through the priority oscillation (some more run through DERP than before, but it works). So the underlying effect does not block deploying DERP.

The logspam is a bit of a problem. I believe it could blow out into being per-packet. So we need to fix that.

**Comment by bradfitz** [comment]
Fixed by @danderson's https://github.com/tailscale/tailscale/commit/2839854994f204a9e95e4d8d410490bb4f25e1fe perhaps?

**Comment by danderson** [comment]
Yes, this was fixed by the patchset in #149. The root bug was: DERP addresses should never be selected as any kind of preferred address by magicsock. DERP is always a last-resort fallback. The destination selection logic had a bug where we could set a DERP address as the peer's roaming endpoint.
